pkg/sql/logictest/tests/local-mixed-24.3/local-mixed-24_3_test: TestLogic_select_for_update failed

pkg/sql/logictest/tests/local-mixed-24.3/local-mixed-24_3_test.TestLogic_select_for_update [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19493472?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19493472?buildTab=artifacts#/) on master @ [3d1f87f368b0f8104ff3ed4a3f47b37ce1cd6b5c](https://github.com/cockroachdb/cockroach/commits/3d1f87f368b0f8104ff3ed4a3f47b37ce1cd6b5c):


```
=== RUN   TestLogic_select_for_update
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/0445a567a274842e2c0e6f00a89fdf71/logTestLogic_select_for_update729671207
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:141: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
[00:58:35] setting distsql_workmem='90961B';
    logic.go:3127: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/31273/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-24.3/local-mixed-24_3_test_/local-mixed-24_3_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/select_for_update:782: SELECT message FROM [SHOW KV TRACE FOR SESSION] WHERE message LIKE 'CPut %%' OR message LIKE 'Put %%' OR message LIKE 'InitPut %%' OR message LIKE 'Del %%' OR message LIKE 'DelRange %%' OR message LIKE 'ClearRange %%' OR message LIKE 'Get %%' OR message LIKE 'Scan %%' 
        expected:
            Scan /Table/113/1/6/{1/1-2/2}
            
        but found (query options: "kvtrace") :
            Scan /Table/113/1/6/{1/1-2/2}
            Scan /Table/113/1/6/2/{1-2}
            
[00:58:35] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/31273/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-24.3/local-mixed-24_3_test_/local-mixed-24_3_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/select_for_update with config local-mixed-24.3: 188 tests, 1 failures
    logic.go:4333: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/31273/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-24.3/local-mixed-24_3_test_/local-mixed-24_3_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/select_for_update:787: error while processing
    logic.go:4333: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/31273/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-24.3/local-mixed-24_3_test_/local-mixed-24_3_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/select_for_update:787: too many errors encountered, skipping the rest of the input
    panic.go:635: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/0445a567a274842e2c0e6f00a89fdf71/logTestLogic_select_for_update729671207
--- FAIL: TestLogic_select_for_update (1.10s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_select_for_update.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-49561

--- a/pkg/sql/logictest/testdata/logic_test/select_for_update
+++ b/pkg/sql/logictest/testdata/logic_test/select_for_update
@@ -1,4 +1,4 @@
-# LogicTest: !weak-iso-level-configs
+# LogicTest: !weak-iso-level-configs !metamorphic-batch-sizes
 # This test assumes that SERIALIZABLE isolation is being used. See
 # select_for_update_read_committed for READ COMMITTED testing.
 

